prusti-viper/src/encoder/builtin_encoder.rs

```diff
@@ -133,6 +150,70 @@ impl BuiltinEncoder {
                     body: None,
                 }
             },
+            BuiltinFunctionKind::SliceLookupPure { slice_elem_ty, return_ty } => {
+                let elem_ty_name = if let vir::Type::TypedRef(ref ty) = slice_elem_ty { ty } else { unreachable!() };
+                let type_pred = format!("Slice${}", elem_ty_name);
```

Comment:
This does not trigger the encoding of the predicate and will probably generate a consistency error like the one in #489. You should call `Encoder::encode_type_predicate_use`, passing the Rust type of the slice. Either you store it in the `BuiltinFunctionKind` (adding a `<'tcx>`) or you build a Rust type on the fly with [`TyCtxt::mk_ty`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.mk_ty).

Reply:
added #499 to pass the rust array type via `PlaceEncoding`, will rebase once that is accepted.